packages/babel-generator/test/fixtures/decorators/decorator-parenthesized-expression-createParenthesizedExpression/output.js

```diff
@@ -15,10 +15,7 @@ class C extends class {} {
       @decs
       @decs.one
       @decs.two()
-      p;
-    }
-    class ShouldAddParens {
-      @((decs[three])())
+      @(decs[three])()
```

Comment:
We add parens here because this syntax is not allowed in decorator versions >= 2021-12:

Here is the current production:
```
Decorator[Yield, Await] :
  @ DecoratorMemberExpression[?Yield, ?Await]
  @ DecoratorParenthesizedExpression[?Yield, ?Await]
  @ DecoratorCallExpression[?Yield, ?Await]

DecoratorMemberExpression[Yield, Await] :
  IdentifierReference[?Yield, ?Await]
  DecoratorMemberExpression[?Yield, ?Await] . IdentifierName
  DecoratorMemberExpression[?Yield, ?Await] . PrivateIdentifier

DecoratorParenthesizedExpression[Yield, Await] :
  ( Expression[+In, ?Yield, ?Await] )

DecoratorCallExpression[Yield, Await] :
  DecoratorMemberExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
```

`@(decs[three])()` is invalid because `DecoratorCallExpression[Yield, Await]` does not allow `DecoratorParenthesizedExpression Arguments`, so we always add parens here so that the output is valid in later decorator proposals.

Does parser currently reject this production when decorator version >= 2021-12?

Reply:
Oh, the parser doesn't accept a decorator version so I was assuming that it is valid. I will revert the behavior change, but leave this as an internal refactor PR to still move the logic to `needsParens`.

Reply:
Well it was indeed valid in earlier decorator versions. The plan looks good to me.